pkg/protocol/engine/booker/inmemorybooker/booker.go

```diff
@@ -102,20 +104,35 @@ func (b *Booker) isPayloadSolid(_ *blocks.Block) (isPayloadSolid bool, err error
 }
 
 func (b *Booker) book(block *blocks.Block) error {
+	if err := b.trackWitnessWeight(block); err != nil {
+		return errors.Wrapf(err, "failed to track witness weight for block %s", block.ID())
+	}
+
+	conflictsToInherit, err := b.inheritConflicts(block)
+	if err != nil {
+		return errors.Wrapf(err, "failed to inherit conflicts for block %s", block.ID())
+	}
+	block.SetConflictIDs(conflictsToInherit)
```

Comment:
you are inheriting both conflicts and payloadconflicts of your parents into your own conflicts. why the difference then? Or are the payload conflicts what your own payload generates? `SetPayloadConflicts` seems unused as of now.

Reply:
`SetPayloadConflicts` needs to be called after the conflicts are determined by the mempool (ie after booking the tx). 

The difference is necessary because of the different references: 
- `Weak`: we only take the payload's conflicts
- `Like`: we take the payload's conflicts + remove everything that is conflicting with it

So storing it separately makes this quite easy.

Reply:
You are inheriting only the conflicts of your parents + your own payload conflict. The former is actually, in turn, their parent's (inherited) conflicts and their own payload conflicts, in a recursive definition. The payload conflicts is what your own payload generates (you are always a conflict if you contain a TX in this case). It probably makes sense to still distinguish the two just to make clear what is the actual origin of the conflict: local (my optional TX payload), or remote (my parents), but certainly you have to set the payload conflict separately to be inherited separately with weak arrows.